sdk/src/cameras/itof-frame/frame_impl.cpp

```diff
@@ -46,35 +46,16 @@
 
 static const int skMetaDataBytesCount = 128;
 
-const std::vector<std::string> availableAttributes = {
-    "mode",       "width",          "height",
-    "subframes",  "embed_width",    "embed_height",
-    "passive_ir", "total_captures", "embed_hdr_length"};
-
 struct FrameImpl::ImplData {
     std::unordered_map<std::string, uint16_t *> m_dataLocations;
     std::shared_ptr<uint16_t[]> m_allData;
     size_t allDataNbBytes;
 };
 
-FrameImpl::FrameImpl() : m_implData(new FrameImpl::ImplData) {
-    for (auto attributeKey : availableAttributes) {
-        m_attributes.emplace(attributeKey, "");
-        if (attributeKey == "embed_hdr_length") {
-            m_attributes[attributeKey] = std::to_string(skMetaDataBytesCount);
-        }
-    }
-}
-
+FrameImpl::FrameImpl() = default;
```

Comment:
I am not sure this is enough. Will a default implementation allocate a new m_implData like the custom copy constructor did?

Reply:
m_implData(new FrameImpl::ImplData){} placed back